File: quick_sort.py
import random

def quick_sort(arr, left, right):
    if left >= right:
        return
    p = right
    less = left-1
    for i in range(left, p):
        if arr[i] < arr[p]:
            less += 1
            arr[i], arr[less] = arr[less], arr[i]
    less += 1
    arr[p], arr[less] = arr[less], arr[p]
    quick_sort(arr, left, less-1)
    quick_sort(arr, less+1, right)

if __name__ == "__main__":
    arr = []
    n = 10
    for i in range(0, n):
        arr.append(random.randint(0, 101))
    print("Unsorted: {}".format(arr))
    quick_sort(arr, 0, n-1)
    print("Sorted: {}".format(arr));

File: quick_sort.c
#include <stdio.h>

void swap(int* a, int* b) {
  int t = *a;
  *a = *b;
  *b = t;
}

void quick_sort(int arr[], int left, int right) {
  int pivot = right;
  int i = left-1;
  if (left >= right) return;

  for (int j = left; j < pivot; ++j) {
    if (arr[j] < arr[pivot]) {
      ++i;
      swap(&arr[i], &arr[j]);
    }
  }
  ++i;
  swap(&arr[i], &arr[pivot]);

  quick_sort(arr, left, i-1);
  quick_sort(arr, i+1, right);
}

int main(int argc, char const *argv[]) {
  int arr[] = {3, 1, 6, 9, 10, 11};
  int n = (int)(sizeof(arr)/sizeof(int));


  quick_sort(arr, 0, n-1);
  for (int i = 0; i < n; ++i) {
    printf((i == n-1) ? "%d\n" : "%d, ", arr[i]);
  }

  return 0;
}
